src/app/shell/admin-tools/platform/platform-info/info-edit/info-edit.component.ts

```diff
@@ -149,16 +170,8 @@ export class InfoEditComponent extends CreateFormComponent implements OnInit, On
 
   onSubmit(): void {
     if (this.PlatformInfoItemArray.valid && this.titleFormControl.valid) {
-      const platformInfoItemArray: CompanyInformationSectionItem[] = [];
-      this.PlatformInfoItemArray.controls.forEach((form: FormGroup) =>
-        platformInfoItemArray.push(new CompanyInformationSectionItem(form.value))
-      );
-
-      const platformInfo = this.editMode
-        ? new CompanyInformation(this.titleFormControl.value, platformInfoItemArray, this.platformInfo.id)
-        : new CompanyInformation(this.titleFormControl.value, platformInfoItemArray);
-
-      this.store.dispatch(new UpdatePlatformInfo(platformInfo, this.platformInfoType));
+      this.isDispatching = true;
+      this.dispatchSubject.next();
```

Comment:
I don't think we need a Subject here

We can modify the method like this and it should be the same.
The same in the other case
```
onSubmit(): void {
    if (this.PlatformInfoItemArray.valid && this.titleFormControl.valid) {
      this.isDispatching = true;
      const platformInfoItemArray: CompanyInformationSectionItem[] = [];
      this.PlatformInfoItemArray.controls.forEach((form: FormGroup) =>
        platformInfoItemArray.push(new CompanyInformationSectionItem(form.value))
      );

      const platformInfo = this.editMode
        ? new CompanyInformation(this.titleFormControl.value, platformInfoItemArray, this.platformInfo.id)
        : new CompanyInformation(this.titleFormControl.value, platformInfoItemArray);

      this.store.dispatch(new UpdatePlatformInfo(platformInfo, this.platformInfoType)).subscribe(() => {
        this.isDispatching = false;
      });;
    }
  }
```

Reply:
In this case we need Subject here, because of one little bug - when you clicking fast on a start of request, it can make one or 2-3 requests max. (only,  but its enough). It's not good for performance and, for example, user get on main page after updating own info. We need to be sure, so I think, throttleTime will be barrier of making unnecessary requests and isDispatching flag will be main functionality.

Reply:
I tried it locally, and in this case it works fine. But maybe I missed something. 
We can use the current solution, it's great anyway. 
Just a few notes: 
- create a variable (for example, `defaultThrottleTime`) instead of a magic number
- move the logic from `switchMap` to a separate method so that we can minimize the code in `ngOnInit`